bin/validate.js

```diff
@@ -113,38 +115,72 @@ function validatePayloadCodecs(vendorId, payloadEncoding) {
     }
   });
 
-  runs.forEach((r) => {
+
+  for (let index = 0; index < runs.length; index++) {
+    const r = runs[index];
     promises.push(
       new Promise((resolve, reject) => {
-        exec(
-          `bin/runscript -codec-path "${r.fileName}" -routine ${r.routine} -input '${JSON.stringify(r.input)}'`,
-          (err, stdout, stderr) => {
-            if (err) {
-              reject(err);
-            } else if (stderr) {
-              reject(stderr);
-            } else {
-              const expected = r.output;
-              let actual = JSON.parse(stdout);
-              if (r.transformOutput) {
-                actual = r.transformOutput(actual);
-              }
-              if (isEqual(expected, actual)) {
-                console.debug(`${r.fileName}:${r.routine}: ${r.description} has correct output`);
-                resolve();
+        setTimeout(async () => {
+          try {
+            const childProcess = spawn('bin/runscript', [
+              '-codec-path',
+              r.fileName,
+              '-routine',
+              r.routine,
+              '-input',
+              JSON.stringify(r.input),
+            ]);
+
+
+            let stdout = '';
+            let stderr = '';
+
+
+            childProcess.stdout.on('data', (data) => {
+              stdout += data.toString();
+            });
+
+
+            childProcess.stderr.on('data', (data) => {
+              stderr += data.toString();
+            });
+
+
+            childProcess.on('error', (error) => {
+              reject(error);
+            });
+
+
+            childProcess.on('close', (code) => {
+              if (code !== 0) {
+                reject(`Process exited with code ${code}\n${stderr}`);
               } else {
-                reject(
-                  `${r.fileName}:${r.routine}: output ${JSON.stringify(actual)} does not match ${JSON.stringify(
-                    expected
-                  )}`
-                );
+                const expected = r.output;
+                let actual = JSON.parse(stdout);
+                if (r.transformOutput) {
+                  actual = r.transformOutput(actual);
+                }
+                if (isEqual(expected, actual)) {
+                  console.debug(`${r.fileName}:${r.routine}: ${r.description} has correct output`);
+                  resolve();
+                } else {
+                  reject(
+                    `${r.fileName}:${r.routine}: output ${JSON.stringify(actual)} does not match ${JSON.stringify(
+                      expected
+                    )}`
+                  );
+                }
               }
-            }
+            });
+          } catch (error) {
+            reject(error);
           }
-        );
+        }, index * 1000); //slow the process 1 second
```

Comment:
So this starts the process later, every second their own process? Why?

Reply:
OK that's a very hacky workaround.

Shouldn't we document using validation for one vendor, with a parameter to `bin/validate.js`? And only use `make validate` in CI?